tokio/tests/signal_info.rs

```diff
@@ -0,0 +1,43 @@
+#![warn(rust_2018_idioms)]
+#![cfg(feature = "full")]
+#![cfg(any(
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "macos",
+    target_os = "netbsd",
+    target_os = "openbsd",
+    target_os = "illumos"
+))]
+#![cfg(not(miri))] // No `sigaction` on Miri
+
+mod support {
+    pub mod signal;
+}
+use support::signal::send_signal;
+
+use tokio::signal;
+use tokio::signal::unix::SignalKind;
+use tokio::sync::oneshot;
+use tokio::time::{timeout, Duration};
+
+#[tokio::test]
+async fn siginfo() {
+    let mut sig = signal::unix::signal(SignalKind::info()).expect("installed signal handler");
+
+    let (fire, wait) = oneshot::channel();
+
+    // NB: simulate a signal coming in by exercising our signal handler
+    // to avoid complications with sending SIGINFO to the test process
+    tokio::spawn(async {
+        wait.await.expect("wait failed");
+        send_signal(libc::SIGINFO);
+    });
```

Comment:
The oneshot channel changes nothing. Tests use the single-threaded runtime, so this spawned task is going to run on this thread, and it will not run until the main test task yields, which it does after polling `sig.recv()` once.

Was that the intent of this code?

Reply:
I copied this from the other tests, specifically: https://github.com/tokio-rs/tokio/blob/c032ea02030b55ef3b16000cd3f51d19009469da/tokio/tests/signal_ctrl_c.rs#L19-L26

I agree that it doesn't make a difference in the narrow case of a single-threaded runtime.

Reply:
Would you like to keep the oneshot channel, have me remove it from this test, or do it yourself across all the signal tests?

Reply:
Filed #7015.